rasa/nlu/training_data/converters/watson_nlu_json_to_yaml_converter.py

```diff
@@ -0,0 +1,157 @@
+                    training_examples, intent, utterance
+                )
+
+        return TrainingData(training_examples, entity_synonyms)
+
+    @staticmethod
+    def _add_training_examples_without_entities(
+        training_examples: List, intent: str, text: str
+    ) -> None:
+        training_examples.append(Message(data={INTENT: intent, TEXT: text}))
+
+    def _unpack_entity_examples(
+        self, text: str, intent: str, training_examples: List, all_entities: List,
+    ) -> List:
+        examples_with_entities = []
+        all_entity_names = set().union(*(d.keys() for d in all_entities))
+        for entity_name in all_entity_names:
+            if entity_name in text:
+                examples_with_entities.append(text)
+                values_of_entity = [
+                    a_dict[entity_name]
+                    for a_dict in all_entities
+                    if entity_name in a_dict
+                ]
+                if values_of_entity is None:
+                    continue
+                for val in values_of_entity[0]:
+                    entities = []
+                    unpack_text = text.replace(entity_name, val.get("value")).replace(
+                        "@", ""
+                    )
+                    start_index = unpack_text.index(val.get("value"))
+                    end_index = start_index + len(val.get("value"))
+                    entities.append(
+                        {
+                            "entity": entity_name,
+                            "start": start_index,
+                            "end": end_index,
+                            "value": val.get("value"),
+                        }
+                    )
+                    training_examples.append(
+                        Message(
+                            data={INTENT: intent, TEXT: unpack_text, ENTITIES: entities}
+                        )
+                    )
+
+        return examples_with_entities
+
+    @staticmethod
+    def _list_all_entities(js: Dict[Text, Any]) -> List:
+        all_entities = []
+        entities = js.get("entities")
+        if entities:
+            for entity in entities:
+                all_entities.append({entity.get("entity"): entity.get("values")})
+        return all_entities
+
+    def _entity_synonyms(self, js: Dict[Text, Any]) -> List:
+        entity_synonyms = []
+        entities = js.get("entities")
+        if entities:
+            for entity in entities:
+                for val in entity.get("values"):
+                    entity_synonyms.append(
+                        {"value": val.get("value"), "synonyms": val.get("synonyms"),}
+                    )
+        return entity_synonyms
+
+    @staticmethod
+    def _check_watson_file(js: Dict[Text, Any]) -> bool:
+        try:
+            if js.get("metadata").get("api_version").get("major_version") == "v2":
+                return True
+            return False
+        except Exception:
```

Comment:
this seems very broad - should we at least a debug statement so that we can see somewhere what happened?